app/controllers/ruleset_controller.rb

```diff
@@ -34,7 +34,24 @@ def create
     # we don't crash out if the API or CLI calls fail.
     begin
       if @ruleset_name == 'security_ruleset' # Call 42Crunch service
-        @crunch_results = Linters::CrunchApi.new.lint_to_json(ruleset_params[:oas_file].tempfile)
+        if ENV['BYPASS_API'] == 'true'
+          crunch_results = File.read('fakecrunchresults.json')
+        else
+          crunch_results = Linters::CrunchApi.new.lint_to_json(ruleset_params[:oas_file].tempfile)
+        end
+        crunch_json = JSON.parse(crunch_results)
+        @score = crunch_json['score']
+        @openapi = crunch_json['openapiState']
+        @counter = crunch_json['issueCounter']
+        @criticality = crunch_json['criticality']
+        @issues = Array.new
+        crunch_json['data']['issues'].each do |issue|
+          lines = Array.new
+          issue[1]['issues'].each do | occurrence|
+            lines.push(occurrence['pointer'])
+          end
+          @issues.push([issue[1]['criticality'],issue[1]['description'], lines])
+        end
```

Comment:
By returning an array of arrays, it is very difficult to determine what the object contains at the point of use (see comment in view). An array of hashes would provide a better result and clearer code.

If `map` is used rather than `each` there is no need to initialize a new array on line 47.

By using [`pluck`](https://api.rubyonrails.org/classes/Enumerable.html#method-i-pluck) the block of code on lines 49-52 can be greatly simplified.

If each `issue` is broken down into its two component parts at the point of iteration (line 48), then it becomes a lot clearer as to what `issue[1]` represents. That is, use the form `hash.each |key, value|` rather than `hash.each |key_value_pair|`

So I would suggest:
```ruby
@issues = crunch_json['data']['issues'].map do |_k, issue|
  {
    pointers: issue['issues'].pluck('pointer'),
    criticality: issue['criticality']
    description: issue['description']
  }
end
```